### PR TITLE
Implement post-checkout Grants Modal Redesign and Refactor

### PR DESCRIPTION
**Pull request by apbendi**
Implements the redesigned "Thank You" modal after grants checkout, with one major limitation:

Currently the image on the modal is an old-school Gitcoin robot, left as a placeholder. The updated design calls for a collections-like image with four quandrants for grant images, plus the user's photo in the middle. However, since the user's cart is not yet a collection at this screen, I don't believe the endpoint for generating these images is set up for this use case. I was wondering if the GPG folks would have some thoughts on how to address this. Alternatively, we could use an updated static image for a version 1.

Screen Shot 2021-02-22 at 4 49 51 PM

Screen Shot 2021-02-22 at 4 50 10 PM

Screen Shot 2021-02-22 at 4 58 53 PM

Screen Shot 2021-02-22 at 4 59 08 PM

### CODE REVIEW

**Comment by apbendi** [comment]
I've rebased the PR to the update `gpg/new-components` branch, resolved linting errors, and updated one of the buttons which was not yet using a proper bootstrap variant. All changes are pushed, however, I'm not able to test locally, because after rebasing, clicking the "Standard Checkout" button fails immediately with the following error:

Screen Shot 2021-02-23 at 9 00 25 PM

File: app/assets/v2/js/grants/funding.js [hunk 1]
```diff
@@ -157,46 +231,49 @@ function sideCartRowForGrant(grant, index) {
...
-    tokenDataList = tokenDataList.filter(token => token.chainId === 58);
+    tokenDataList = tokenDataList.filter((token) => token.chainId === 58);
     tokenDefault = 'DOT';
   } else if (grant.tenants && grant.tenants.includes('KUSAMA')) {
-    tokenDataList = tokenDataList.filter(token => token.chainId === 59);
+    tokenDataList = tokenDataList.filter((token) => token.chainId === 59);
     tokenDefault = 'KSM';
   } else {
-    tokenDataList = tokenDataList.filter(token => token.chainId === 1);
+    tokenDataList = tokenDataList.filter((token) => token.chainId === 1);
```
Comment (thelostone-mc) [inline app/assets/v2/js/grants/funding.js:270]: ^ could we undo these @apbendi

File: app/assets/v2/js/grants/funding.js [hunk 2]
```diff
@@ -205,8 +282,8 @@ function tokenOptionsForGrant(grant) {
             <option value="${tokenDefault}">${tokenDefault}</option>
         `;
 
-    tokenDataList = tokenDataList.filter(tokenData => {
-      return (tokenData.address === grant.grant_token_address);
+    tokenDataList = tokenDataList.filter((tokenData) => {
```
Comment (thelostone-mc) [inline app/assets/v2/js/grants/funding.js:285]: this change is not needed

File: app/assets/v2/js/grants/funding.js (hunk 1 above)
Comment (apbendi) [reply app/assets/v2/js/grants/funding.js:270]: hey @thelostone-mc, I believe I made this change due to linting errors, but happy to undo if that's not an issue.

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**